docs/faq.md

```diff
@@ -0,0 +1,151 @@
+# FAQ
+This is more meant as a "Fully Anticipated Questions" than a "Frequently Asked Questions", as they're anticipated as common pitfalls to using NDE.
+
+## I am new to Nix. Is this the only way to use Doom Emacs with Nix/NixOS?
+
+Nope! Doom Emacs is still perfectly usable imperatively. In fact, the very author of Doom Emacs uses NixOS and install Doom Emacs with an ["imperative" setup](https://github.com/hlissner/dotfiles/blob/master/modules/editors/emacs.nix). You could just follow the instructions on the [Doom Emacs GitHub repository](https://github.com/doomemacs/doomemacs) to get a working setup.
+
+## OK, I put your snippets into my NixOS configuration, and I put my Doom Emacs configuration as well. How do I `doom sync`?
+
+To update your Doom Emacs config, you simply rebuild your configuration. For example, in NixOS you can use `nixos-rebuild switch` (or `home-manager switch` if you use Home-Manager standalone). nix-doom-emacs will do everything else for you.
+
+## How do I install my favourite package?
+
+You should use the `emacsPackagesOverlay` attribute. Here's an example that installs `magit-delta`, which depends on Git:
+
+```nix
+programs.doom-emacs = {
+  # ...
+  emacsPackagesOverlay = self: super: {
+    magit-delta = super.magit-delta.overrideAttrs (esuper: {
+      buildInputs = esuper.buildInputs ++ [ pkgs.git ];
+    });
+  }
+};
+```
+
+### But what if my package is from a Git source, and isn't on ELPA?
+
+This requires a few more lines, but it isn't by any means difficult. For an example, this installs `idris2-mode` which isn't on ELPA, but is hosted on GitHub:
+
+```nix
+programs.doom-emacs = {
+  # ...
+  emacsPackagesOverlay = self: super: {
+    idris2-mode = self.trivialBuild {
+      pname = "idris2-mode";
+      ename = "idris2-mode";
+      version = "0.0.0";
+      buildInputs = [ self.prop-menu ];
+      src = pkgs.fetchFromGitHub {
+        owner = "idris-community";
+        repo = "idris2-mode";
+        rev = "4a3f9cdb1a155da59824e39f0ac78ccf72f2ca97";
+        sha256 = "sha256-TxsGaG2fBRWWP9aas59kiNnUVD4ZdNlwwaFbM4+n81c=";
+      };
+    };
+  }
+};
+```
+
+### Help! My favourite package doesn't work after adding it in!
```

Comment:
"doesn't work after adding it in" is extremely generic and yet this addresses a specific issue that's already been mentioned before a bit. Maybe drop this whole question? It's kind of implied by the user having knowledge of Nix I think. I'm not sure about our scope though.